test/ConstantProduct/DeploymentParamsTest.sol

```diff
@@ -40,7 +40,7 @@ abstract contract DeploymentParamsTest is ConstantProductTestHarness {
         mockZeroAllowance(token, expectedDeploymentAddress(), spenderBadApproval);
         vm.mockCallRevert(
             address(token),
-            abi.encodeCall(IERC20.approve, (spenderBadApproval, type(uint256).max)),
+            abi.encodeWithSelector(IERC20.approve.selector, spenderBadApproval),
```

Comment:
**notice**: making the `approve` call generically revert when called with `spenderBadApproval` (reverts for every amount, instead of for `type(uint256).max`)

Reply:
```suggestion
            // Notice: we intentionally don't match the approved amount.
            abi.encodeWithSelector(IERC20.approve.selector, spenderBadApproval),
```